sys/net/routing/nhdp/iib_table.c

```diff
@@ -216,6 +234,138 @@ void iib_propagate_nb_entry_change(nib_entry_t *old_entry, nib_entry_t *new_entr
     }
 }
 
+void iib_process_metric_msg(iib_link_set_entry_t *ls_entry, uint64_t int_time)
+{
+    switch (NHDP_METRIC) {
+#if (NHDP_METRIC == NHDP_LMT_HOP_COUNT)
+        case NHDP_LMT_HOP_COUNT:
+            /* Hop metric value for an existing direct link is always 1 */
+            ls_entry->metric_in = 1;
+            ls_entry->metric_out = 1;
+            if (ls_entry->nb_elt) {
+                ls_entry->nb_elt->metric_in = 1;
+                ls_entry->nb_elt->metric_out = 1;
+            }
+            break;
+#endif
+#if (NHDP_METRIC == NHDP_LMT_DAT)
+        case NHDP_LMT_DAT:
+            /* Process required DAT metric steps */
+            ls_entry->hello_interval = rfc5444_timetlv_encode(int_time);
+            if (ls_entry->last_seq_no == 0) {
+                timex_t now, i_time;
+                vtimer_now(&now);
+                i_time = timex_from_uint64(int_time * MS_IN_USEC * DAT_HELLO_TIMEOUT_FACTOR);
+                ls_entry->dat_received[0]++;
+                ls_entry->dat_total[0]++;
+                ls_entry->dat_time = timex_add(now, i_time);
+            }
+            break;
+#endif
+        default:
+            /* NHDP_METRIC is not set properly */
+            DEBUGF("[WARNING] Unknown NHDP_METRIC setting\n");
+            break;
+    }
+}
+
+void iib_process_metric_pckt(iib_link_set_entry_t *ls_entry, uint32_t metric_out, uint16_t seq_no)
+{
+    switch (NHDP_METRIC) {
+#if (NHDP_METRIC == NHDP_LMT_HOP_COUNT)
+        case NHDP_LMT_HOP_COUNT:
+            /* Nothing to do here */
+            break;
+#endif
+#if (NHDP_METRIC == NHDP_LMT_DAT)
+        case NHDP_LMT_DAT:
+            /* Metric packet processing */
+            if (ls_entry->last_seq_no == 0) {
+                ls_entry->dat_received[0] = 1;
+                ls_entry->dat_total[0] = 1;
+            }
+            else {
+                /* Don't add values to the queue for duplicate packets */
+                if (seq_no != ls_entry->last_seq_no) {
+                    uint16_t seq_diff;
+                    if (seq_no < ls_entry->last_seq_no) {
+                        seq_diff = (uint16_t) ((((uint32_t) seq_no) + 0xFFFF) - ls_entry->last_seq_no);
```

Comment:
This would result in values that are not treated correctly by the restart detection.

The DAT draft states:
>   diff_seqno(new, old)  - an operation which returns the positive
      distance between two elements of the circular sequence number
      space defined in section 5.1 of [RFC5444].  Its value is either
      (new - old) if this result is positive, or else its value is (new
      - old + 65536).

Of course I could adapt the restart detection, but this would result in adding complexity somewhere else.